src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -215,6 +196,105 @@ int halide_hexagon_remote_power_hvx_on() {
     return 0;
 }
 
+int halide_hexagon_remote_power_hvx_on_mode(int mode) {
+    int set_mips = 0;
+    unsigned int mipsPerThread = 0;
+    unsigned int mipsTotal = 0;
+    int set_bus_bw = 0;
+    uint64_t bwBytePerSec = 0;
+    unsigned int busbwUsagePercentage = 0;
+    int set_latency = 0;
+    int latency = 0;
+    int result = 0;
+
+    // Only need to compute perf parameters if power_ref_count is zero
```

Comment:
I think maybe it would be simpler/safer to just compute the parameters every time, and I doubt it would impact performance much, since this function will not be called by generated code inside the halide pipelines (it will only be called manually). It might also enable further cleanup (comment below).

I have lofty dreams of making the reference count thread safe at some point, in which case, this logic is dangerous.